File: src/main/java/com/simonvonxcvii/turing/model/dto/UserDTO.java
package com.simonvonxcvii.turing.model.dto;

import com.simonvonxcvii.turing.model.query.PageQuery;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.time.LocalDateTime;
import java.util.Collection;
import java.util.List;

@Accessors(chain = true)
@Getter
@Setter
public class UserDTO extends PageQuery {
    /**
     * 用户姓名
     */
    @NotBlank(message = "用户姓名不能为空")
    public String name;
    /**
     * 用户手机号
     */
    @NotBlank(message = "用户手机号不能为空")
    public String mobile;
    /**
     * 用户性别
     */
    @NotBlank(message = "用户性别不能为空")
    public String gender;
    /**
     * 单位 id
     */
    @NotBlank(message = "单位 id 不能为空")
    public Integer orgId;
    /**
     * 单位名称
     */
    public String orgName;
    /**
     * 部门
     */
    public String department;
    /**
     * 登录账号
     */
    @NotBlank(message = "登录账号不能为空")
    public String username;
    /**
     * 是否已过期
     */
    public Boolean accountNonExpired;
    /**
     * 是否已锁定
     */
    public Boolean accountNonLocked;
    /**
     * 是否凭证已过期
     */
    public Boolean credentialsNonExpired;
    /**
     * 是否启用
     */
    public Boolean enabled;
    /**
     * 是否单位管理员
     */
    public Boolean manager;
    /**
     * 是否需要重新设置密码
     */
    public Boolean needSetPassword;
    /**
     * 用户角色
     */
    public Collection<RoleDTO> authorities;
    /**
     * 用户 id
     */
    private Integer id;
    /**
     * 创建日期
     */
    private LocalDateTime createdDate;
    /**
     * 角色集合
     */
    @NotEmpty(message = "角色集合不能为空")
    private List<Integer> roleList;
}

File: src/main/java/com/simonvonxcvii/turing/model/dto/UserDTO.kt
package com.simonvonxcvii.turing.model.dto

import com.simonvonxcvii.turing.model.query.PageQuery
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.NotNull
import java.io.Serializable
import java.time.LocalDateTime

/**
 * DTO for [com.simonvonxcvii.turing.entity.User]
 *
 * @author Simon Von
 * @since 12/30/2022 4:03 PM
 */
data class UserDTO(
    /**
     * 用户姓名
     */
    @field:NotBlank(message = "用户姓名不能为空")
    var realName: String? = null,

    /**
     * 用户手机号
     */
    @field:NotNull(message = "用户手机号不能为空")
    var mobile: Long? = null,

    /**
     * 用户性别
     */
    @field:NotBlank(message = "用户性别不能为空")
    var gender: String? = null,

    /**
     * 单位 id
     */
    @field:NotBlank(message = "单位 id 不能为空")
    var orgId: Int? = null,

    /**
     * 单位名称
     */
    var orgName: String? = null,

    /**
     * 部门
     */
    var department: String? = null,

    /**
     * 登录账号
     */
    @field:NotBlank(message = "登录账号不能为空")
    var username: String? = null,

    /**
     * 是否已过期
     */
    var accountNonExpired: Boolean? = null,

    /**
     * 是否已锁定
     */
    var accountNonLocked: Boolean? = null,

    /**
     * 是否凭证已过期
     */
    var credentialsNonExpired: Boolean? = null,

    /**
     * 是否启用
     */
    var enabled: Boolean? = null,

    /**
     * 是否单位管理员
     */
    var manager: Boolean? = null,

    /**
     * 是否需要重新设置密码
     */
    var needResetPassword: Boolean? = null,

    /**
     * 用户角色
     */
    var authorities: MutableCollection<RoleDTO>? = mutableListOf(),

    /**
     * 用户角色编码
     */
    var roles: MutableSet<String>? = mutableSetOf(),

    /**
     * 用户 id
     */
    var id: Int? = null,

    /**
     * 创建时间
     */
    var createdDate: LocalDateTime? = null,

    /**
     * 角色集合
     */
    @field:NotEmpty(message = "角色集合不能为空")
    var roleIdList: MutableList<Int>? = mutableListOf()
) : Serializable, PageQuery()
